lib/websocket.c

```diff
@@ -57,11 +57,22 @@ static void on_recv(h2o_socket_t *sock, const char *err)
 static void on_write_complete(h2o_socket_t *sock, const char *err)
 {
     h2o_websocket_conn_t *conn = sock->data;
+    h2o_iovec_t *buf;
+    size_t i;
 
     if (err != NULL) {
         on_close(conn);
         return;
     }
+    assert(conn->_write_buf.cnt == 0);
+    for (i = 0; i < sizeof(conn->_write_buf.bufs) / sizeof(conn->_write_buf.bufs[0]); ++i) {
+        buf = &conn->_write_buf.bufs[i];
+        if (buf->base != NULL) {
```

Comment:
How about doing `if (buf->base == NULL) break;` instead? I think that all entries following one that contains `NULL` are guaranteed to be `NULL`.

Reply:
yes, fixed.